Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp

```diff
@@ -726,6 +726,7 @@ WebIDL::ExceptionOr<void> set_up_readable_stream_default_controller_from_underly
     // 5. If underlyingSourceDict["start"] exists, then set startAlgorithm to an algorithm which returns the result of invoking underlyingSourceDict["start"] with argument list « controller » and callback this value underlyingSource.
     if (underlying_source.start) {
         start_algorithm = [&, start = underlying_source.start]() -> WebIDL::ExceptionOr<JS::GCPtr<WebIDL::Promise>> {
+            // FIXME: This needs to create a rejected Promise for throw completions, which invoke_callback() currently doesn't do.
```

Comment:
I'm pretty sure these functions currently propagating errors when they should turn them into rejected promises is one of the major reasons why so many `!` AO invocations still need a `TRY()`.

Reply:
Right, if I'm understanding this correctly, `clean_up_on_return` will always return a normal completion with either a rejected promise or a non-promise value. But because we just always return the completion, it may hold an error so we need to `TRY` (which still isn't correct because there is no rejected promise being created). So according to the spec, that error shouldn't be propogated, even at the call site of the algorithm ([here](https://github.com/SerenityOS/serenity/blob/19db57728fbb307daf32289ed2337b9642956ae1/Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp#L515), for example). 

That's tricky, I definitely would not have noticed this 😅

Reply:
> So according to the spec, that error shouldn't be propogated, even at the call site of the algorithm (here, for example).

Yep, that's assumed to always return a promise, never an error (ignoring OOM) so `Upon rejection of pullPromise with reason e` would apply.